src/applets/budgie-menu/BudgieMenuWindow.vala

```diff
@@ -74,7 +74,11 @@ public class MenuButton : Gtk.Button
         var img = new Gtk.Image.from_gicon(parent.get_icon(), Gtk.IconSize.INVALID);
         img.pixel_size = icon_size;
         img.margin_end = 7;
-        var lab = new Gtk.Label(parent.get_display_name());
+        string menu_label = parent.get_display_name () ;
+         if( menu_label.length > 28 ){
```

Comment:
Why not use the actually Gtk ellipse functions? And I'm pretty sure we ellipsize later on - this is going to really break usability.